src/components/Swap/Settings/index.tsx

```diff
@@ -1,72 +1,44 @@
-import { settingsAtom } from 'state/swap/settings'
+import { useState } from 'react'
 import styled from 'styled-components/macro'
-import { ThemedText } from 'theme'
 
-import { IconButton, TextButton } from '../../Button'
+import { IconButton } from '../../Button'
 import Column from '../../Column'
-import Dialog, { Header } from '../../Dialog'
-import { BoundaryProvider } from '../../Popover'
+import Popover, { BoundaryProvider } from '../../Popover'
 import MaxSlippageSelect from './MaxSlippageSelect'
 import TransactionTtlInput from './TransactionTtlInput'
 
-export function SettingsDialog() {
+export function SettingsPopover() {
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
-  const { onSettingsReset } = useAtomValue(swapEventHandlersAtom)
-  const resetSettingsBase = useResetAtom(settingsAtom)
-  const resetSettings = useCallback(() => {
-    onSettingsReset?.()
-    resetSettingsBase()
-  }, [onSettingsReset, resetSettingsBase])
+
   return (
-    <>
-      <Header title={<Trans>Settings</Trans>} ruled>
-        <TextButton onClick={resetSettings}>
-          <ThemedText.ButtonSmall>
-            <Trans>Reset</Trans>
-          </ThemedText.ButtonSmall>
-        </TextButton>
-      </Header>
-      <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
-        <BoundaryProvider value={boundary}>
-          <MaxSlippageSelect />
-          <TransactionTtlInput />
-        </BoundaryProvider>
-      </Column>
-    </>
+    <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
+      <BoundaryProvider value={boundary}>
+        <MaxSlippageSelect />
+        <TransactionTtlInput />
+      </BoundaryProvider>
+    </Column>
   )
 }
 
-const SettingsButton = styled(IconButton)<{ hover: boolean }>`
-  ${SettingsIcon} {
-    transform: ${({ hover }) => hover && 'rotate(45deg)'};
-    transition: ${({ hover }) => hover && 'transform 0.25s'};
-    will-change: transform;
+const SettingsButton = styled(IconButton)`
+  ${SettingsIcon}:hover {
+    transform: rotate(45deg);
+    transition: transform 0.25s;
   }
 `
 
-export default function Settings({ disabled }: { disabled?: boolean }) {
+export default function Settings() {
   const [open, setOpen] = useState(false)
-  const [hover, setHover] = useState(false)
+  const [wrapper, setWrapper] = useState<HTMLDivElement | null>(null)
+
   return (
-    <>
-      <SettingsButton
-        disabled={disabled}
-        hover={hover}
-        onClick={() => setOpen(true)}
-        onMouseEnter={() => setHover(true)}
-        onMouseLeave={() => setHover(false)}
-        icon={SettingsIcon}
-      />
-      {open && (
-        <Dialog color="module" onClose={() => setOpen(false)}>
-          <SettingsDialog />
-        </Dialog>
-      )}
-    </>
+    <div ref={setWrapper}>
+      <BoundaryProvider value={wrapper}>
+        <Popover showArrow={false} offset={10} show={open} placement="top-end" content={<SettingsPopover />}>
```

Comment:
Past usage of Popover was for tooltip and tooltip-like usages, so they are missing some functionality necessary for the more Dialog-like Popover.

Specifically, the Popover should dismiss both when you hit <ESC> and when you click off of it. See the Dialog for an example implementation.

Reply:
seems more appropriate to convert the existing `Popover` than trying to coalesce Dialog into this new functionality though. Popover is essentially the same except you have different enter/exit states

Reply:
asked in figma about dismissing behavior

Reply:
Feels like terrible UX to me to only be able to dismiss the settings popover by clicking the cog icon again. I feel pretty strongly that it should retain the Dialog exit behaviors.